runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java

```diff
@@ -0,0 +1,234 @@
+	{
+		return configManager.getConfig(BankConfig.class);
+	}
+
+	@Subscribe
+	public void checkBankState(ConfigChanged event)
+	{
+		if (!config.isEnabled() || client.getWidget(WidgetInfo.BANK_ITEM_COUNT) == null) return;
+
+		CancelTasks();
+
+		taskWidget = executor.submit(() -> WaitForWidget(WidgetInfo.BANK_ITEM_COUNT, Double.MAX_VALUE));
+
+		return;
+	}
+
+	@Subscribe
+	public void onItemChanged(MenuOptionClicked event)
+	{
+		String strMenuOption = event.getMenuOption();
+
+		if (strMenuOption.contains("Withdraw") || strMenuOption.contains("Deposit") || strMenuOption.contains("View tab") || strMenuOption.contains("View all") || strMenuOption.contains("Bank"))
+		{
+			CancelTasks();
+
+			double dblTick = System.currentTimeMillis() + 10000;
+
+			taskWidget = executor.scheduleAtFixedRate(() -> WaitForWidget(WidgetInfo.BANK_ITEM_COUNT, dblTick), 0, 1, TimeUnit.SECONDS);
+		}
+
+		return;
+	}
+
+	private void WaitForWidget(WidgetInfo info, double dblTick)
+	{
+		if (dblTick < System.currentTimeMillis() || !config.isEnabled())
+		{
+			taskWidget.cancel(true);
+			return;
+		}
+
+		Widget w = client.getWidget(info);
+
+		if (w != null)
+		{
+			taskWidget.cancel(true);
+			futureCalculateBank = executor.schedule(runCalculateBank, 1, TimeUnit.SECONDS);
+		}
+	}
+
+	private void CancelTasks()
+	{
+		if (taskWidget != null)
+		{
+			taskWidget.cancel(true);
+			taskWidget = null;
+		}
+
+		if (futureCalculateBank != null)
+		{
+			runCalculateBank.abort();
+			futureCalculateBank.cancel(false);
+			futureCalculateBank = null;
+		}
+	}
+
+	private int[] getItemPrices(int intID, int quantity)
+	{
+		if (ITEM_COINS == intID) return new int[]{quantity, quantity};
+
+		ItemComposition itemComposition = itemManager.getItemComposition(intID);
+
+		if (itemComposition == null) return new int[]{0, 0};
+
+		log.warn(itemComposition.getName());
+
+		ItemPrice itemPrice;
+		try
+		{
+			itemPrice = itemManager.getItemPrice(itemComposition.getId());
+		} catch (IOException e)
+		{
+			log.warn("Error looking up item price", e);
+			return new int[]{0, 0};
+		}
+
+		int itemCompositionPrice = itemComposition.getPrice();
+
+		return new int[]{itemPrice == null ? 0 : itemPrice.getPrice() * quantity, itemCompositionPrice <= 0 ? 0 : Math.round(itemCompositionPrice * HIGH_ALCHEMY_CONSTANT) * quantity};
+	}
+
+	private java.util.List<Widget> GetItems()
+	{
+		java.util.List<Widget> widgetItems = new ArrayList<Widget>();
+		Widget widgetItemContainer = client.getWidget(WidgetInfo.BANK_ITEM_CONTAINER);
+
+		if (widgetItemContainer == null) return widgetItems;
+
+		Widget[] widgetChildren = widgetItemContainer.getDynamicChildren();
+
+		for (int i = 0; i < widgetChildren.length; i++)
+		{
+			if (widgetChildren[i].getItemId() == ITEM_EMPTY || widgetChildren[i].getItemId() <= 0 || widgetChildren[i].isHidden())
+				continue;
+
+			widgetItems.add(widgetChildren[i]);
+		}
+
+		return widgetItems;
+	}
+
+	private String formatNumber(double n)
```

Comment:
Both formatNumber methods should be static

Reply:
Also, use the ItemManager#quantityToStackSize instead (and modify that method to include also bills etc.